### ISSUE TITLE
Add Layout/MultilineMethodCallIndentation option for e.g. `indented_relative_to_first_statement`

### ISSUE THREAD

**Issue by pdobb** [rubocop-hq/rubocop#8072]
## Is your feature request related to a problem? Please describe.

I have had the [Layout/MultilineOperationIndentation](https://docs.rubocop.org/rubocop/cops_layout.html#layoutmultilineoperationindentation) cop disabled since I started using Rubocop years ago because the existing options are insufficient for my taste. But I need to start using a cop such as this because I spend too much time in peer review suggesting unenforceable indentation conventions for multiline operation indentation.

Basically: I want an enforceable convention that uses indentation to show line wrapping at the first part of the statement that isn't a an assignment operator or control flow operator.

## Describe the solution you'd like

I would like an option similar to [indented_relative_to_receiver](https://docs.rubocop.org/rubocop/cops_layout.html#enforcedstyle-indented_relative_to_receiver) from the [Layout/MultilineMethodCallIndentation](https://docs.rubocop.org/rubocop/cops_layout.html#layoutmultilinemethodcallindentation) cop.

Fro example, a new option such as `indented_relative_to_first_statement` that would enforce:

```ruby
# Bad
my_var =  my_condition1 &&
  my_condition2 &&
  my_condition3

# Good
my_var =  my_condition1 &&
           my_condition2 &&
           my_condition3

###

# Bad
my_var =
  my_condition1 &&
  my_condition2
  my_condition3

# Good
my_var =
  my_condition1 &&
    my_condition2 &&
    my_condition3

### 

# Bad
return if my_condition1 &&
  my_condition2
  my_condition3

# Good
return if my_condition1 &&
            my_condition2
            my_condition3
```

## Describe alternatives you've considered

I've tried using the `indented` option:
```yaml
Layout/MultilineOperationIndentation:
  EnforcedStyle: indented
  IndentationWidth: 2 # I'm not sure exactly what this does...
```

And while I can accept this style:

```ruby
# OK
my_var =
  my_condition1 &&
  my_condition2
```

I cannot accept this style because I don't think it's clear that `my_condition2` is a continuation of the `&&` statement versus being a continuation of the `return if` control flow operator:

```ruby
# Bad (IMO)
return if my_condition1 &&
  my_condition2
```

## Additional context

Ideally, the indentation level is only enforced on the first line-wrap. The goal being to avoid infinite stair steps:

```ruby
# Bad
my_var =
  my_condition1 &&
    my_condition2 &&
      my_condition3 &&
        my_condition4 &&
          # ...

# Bad
my_var =
  my_condition1 &&
    my_condition2 &&
    my_condition3 &&
    my_condition4 &&
    # ...
```

**Comment by pdobb** [comment]
Just wanted to bump this issue up. Still having the same want in 2025. Thanks.

**Comment by pdobb** [comment]
@koic I'm not sure how to bump this up but have been hurting a bit without a solution here. Would you be able to take a look? If not interested, I understand. Thanks either way 👍

**Comment by roberts1000** [comment]
I would also like to see this enhancement. `indented` in each cop behaves differently. The cop is similar to `MultilineMethodCallIndentation` so  it makes sense for the options to be "aligned". 😀
